src/platform/intel/cavs/lib/clk.c

```diff
@@ -61,6 +61,29 @@ static inline void select_cpu_clock(int freq_idx, bool release_unused)
 }
 #endif
 
+static inline void select_cpu_clock(int freq_idx, bool release_unused)
+{
+	struct clock_info *clk_info = clocks_get();
+	int flags[PLATFORM_CORE_COUNT];
+	int i;
+
+	/* lock clock for all cores */
+	for (i = 0; i < PLATFORM_CORE_COUNT; i++)
+		spin_lock_irq(&clk_info[CLK_CPU(i)].lock, flags[i]);
+
+	/* change clock */
+	select_cpu_clock_hw(freq_idx, release_unused);
+	for (i = 0; i < PLATFORM_CORE_COUNT; i++)
+		clk_info[CLK_CPU(i)].current_freq_idx = freq_idx;
+
+	/* unlock clock for all cores */
+	for (i = 0; i < PLATFORM_CORE_COUNT; i++)
+		spin_unlock_irq(&clk_info[CLK_CPU(i)].lock, flags[i]);
+
```

Comment:
These should be released in reverse order of aquistion. e.g.
lock 0,1,2,3
unlock 3,2,1,0
This will help to avoid deadlock

Reply:
updated